mono/eglib/goutput.c

```diff
@@ -126,12 +126,23 @@ g_log (const gchar *log_domain, GLogLevelFlags log_level, const gchar *format, .
 	va_end (args);
 }
 
+static char *failure_assertion = NULL;
+
+const char *
+g_get_assertion_message (void)
+{
+	return failure_assertion;
+}
+
 void
 g_assertion_message (const gchar *format, ...)
 {
 	va_list args;
 
 	va_start (args, format);
+
+	g_vasprintf (&failure_assertion, format, args);
```

Comment:
This breaks assert. Put va_start after it. Or sprintf into a local and use it for both (g_log ("%s", foo)).

Reply:
https://github.com/mono/mono/pull/8727/commits/a1baf487d073f6d66d11ac0134f2fa6b0e06408e fixes this.